app/src/component/channelSelector/view.js

```diff
@@ -0,0 +1,243 @@
+      this.setState({ addingChannel: true });
+      if (onChannelChange) {
+        onChannelChange(channel);
+      }
+      this.handleNewChannelBidChange(newChannelBid);
+    } else {
+      this.setState({ addingChannel: false });
+      if (onChannelChange) {
+        onChannelChange(channel);
+      }
+    }
+  };
+
+  handleNewChannelNameChange = value => {
+    const { notify } = this.props;
+
+    let newChannelName = value;
+
+    if (newChannelName.startsWith('@')) {
+      newChannelName = newChannelName.slice(1);
+    }
+
+    let newChannelNameError;
+    if (newChannelName.length > 1 && !isNameValid(newChannelName.substr(1), false)) {
+      notify({ message: 'LBRY channel names must contain only letters, numbers and dashes.' });
+    }
+
+    this.setState({
+      newChannelNameError,
+      newChannelName,
+    });
+  };
+
+  handleNewChannelBidChange = newChannelBid => {
+    const { balance, notify } = this.props;
+    let newChannelBidError;
+    if (newChannelBid === 0) {
+      newChannelBidError = __('Your deposit cannot be 0');
+    } else if (newChannelBid === balance) {
+      newChannelBidError = __('Please decrease your deposit to account for transaction fees');
+    } else if (newChannelBid > balance) {
+      newChannelBidError = __('Deposit cannot be higher than your balance');
+    }
+
+    notify({ message: newChannelBidError });
+
+    this.setState({
+      newChannelBid,
+      newChannelBidError,
+    });
+  };
+
+  handleCreateChannelClick = () => {
+    const { balance, createChannel, onChannelChange, notify } = this.props;
+    const { newChannelBid, newChannelName } = this.state;
+
+    if (newChannelName.trim().length === 0 || !isNameValid(newChannelName.substr(1), false)) {
+      notify({ message: 'LBRY channel names must contain only letters, numbers and dashes.' });
+      return;
+    }
+
+    if (this.channelExists(newChannelName)) {
+      notify({ message: 'You have already created a channel with the same name.' });
+      return;
+    }
+
+    if (newChannelBid > balance) {
+      notify({ message: 'Deposit cannot be higher than your balance' });
+      return;
+    }
+
+    const channelName = `@${newChannelName}`;
+
+    this.setState({
+      creatingChannel: true,
+      createChannelError: undefined,
+    });
+
+    const success = () => {
+      this.setState({
+        creatingChannel: false,
+        addingChannel: false,
+        currentSelectedValue: channelName,
+        showCreateChannel: false,
+      });
+
+      if (onChannelChange) {
+        onChannelChange(channelName);
+      }
+    };
+
+    const failure = () => {
+      notify({ message: 'Unable to create channel due to an internal error.' });
+      this.setState({
+        creatingChannel: false,
+      });
+    };
+
+    createChannel(channelName, newChannelBid).then(success, failure);
+  };
+
+  channelExists = name => {
+    const { channels = [] } = this.props;
+    for (let channel of channels) {
```

Comment:
It's not strictly necessary, but this can be done functionally:

```
return channels.reduce((pass, channel) => 
  return pass || name.toLowerCase() === channel.name.toLowerCase() ||  `@${name}`.toLowerCase() === channel.name.toLowerCase();
);
```